packages/mask/background/services/wallet/services/wallet/index.ts

```diff
@@ -68,6 +77,28 @@ export async function getWalletPrimary() {
     )
 }
 
+export async function getMnemonicId(storedKeyInfo: api.IStoredKeyInfo) {
+    const masterPassword = await password.INTERNAL_getMasterPasswordRequired()
+    const firstWallet = await Mask.createAccountOfCoinAtPath({
+        coin: api.Coin.Ethereum,
+        password: masterPassword,
+        derivationPath: `${HD_PATH_WITHOUT_INDEX_ETHEREUM}/0`,
+        StoredKeyData: storedKeyInfo.data,
+    })
+    if (!firstWallet?.account?.address) return
+    return firstWallet?.account?.address
+}
+
+export async function getMnemonicIdByWords(mnemonic: string) {
```

Comment:
Two much query methods. Please store the `mnemonicId` in `sha3(words)`. It's computable everywhere and enough with query method `getWalletByMnemonicId()`.